src/Telegram.php

```diff
@@ -1285,4 +1284,56 @@ public function getUpdateFilter(): ?callable
     {
         return $this->update_filter;
     }
+
+    /**
+     * Converts the name of a class into the name of a command.
+     *
+     * @param string $class For example FooBarCommand
+     *
+     * @return string for example foo_bar. In case of errors, returns an empty string
+     */
+    protected function classNameToCommandName(string $class): string
+    {
+        if (!preg_match('/^(.+)Command$/', $class, $matches)) {
+            return '';
+        }
+        $temp = $matches[1];
+        $chunks = [];
+        $currentUpperCaseLetter = '';
+        while ($temp !== '') {
+            if (!preg_match('/\p{Lu}/u', $temp, $match, PREG_OFFSET_CAPTURE)) {
+                break;
+            }
+            // $match[0][0] contains first upper case character
+            // $match[0][1] contains the start position (in bytes) of the first upper case character
+            [$upperCaseLetter, $upperCaseLetterOffset] = $match[0];
+            if ($upperCaseLetterOffset > 0) {
+                $chunks[] = $currentUpperCaseLetter . substr($temp, 0, $upperCaseLetterOffset);
+            } elseif ($currentUpperCaseLetter !== '') {
+                $chunks[] = $currentUpperCaseLetter;
+            }
+            $temp = substr($temp, $upperCaseLetterOffset + strlen($upperCaseLetter));
+            $currentUpperCaseLetter = $upperCaseLetter;
+        }
+        $lastChunk = $currentUpperCaseLetter . $temp;
+        if ($lastChunk !== '') {
+            $chunks[] = $lastChunk;
+        }
+        return implode('_', array_map('mb_strtolower', $chunks));
```

Comment:
The whole method is pretty verbose for something that could literally be something like 
```
mb_strtolower(preg_replace('/(.)(?=[\p{Lu}])/u', '$1_', $value));
```

But we could rewrite that after merging.

Reply:
Yep, much better!